lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -447,6 +453,51 @@ spec = do
                             . slotStartTime sps
                     Just slot === f slot
 
+    describe "Converting Range SlotId to Range UTCTime and back" $ do
+        let f sps r = fmap (slotStartTime sps) <$> slotRange sps r
+        let sharedDescription = " (where f = fmap slotStartTime . slotRange)"
+        it ("f timeRange ⊆ timeRange" ++ sharedDescription) $
+            withMaxSuccess 1000 $ property $
+                \sps timeRange -> let r = getUniformTime <$> timeRange in
+                    cover 10 (isJust $ f sps r) "slot range exists" $ do
+                    case f sps r of
+                        Just r' -> r' `isSubrangeOf` r
+                        Nothing -> True
+        it ("f^n === f" ++ sharedDescription) $
+            withMaxSuccess 1000 $ property $
+                \sps timeRange (NonNegative n) -> do
+                    let r = getUniformTime <$> timeRange
+                    let r' = f sps r
+                    iterate (>>= f sps) r' !! n === r'
+
+        it "slots in `slotRange [t1, t3]` are also in one or more of \
+           \`slotRange [t1, t2]` or `slotRange [t2, t3]` if t1 < t2 < t3" $
```

Comment:
Regarding the following property:

> "slots in `slotRange [t1, t3]` are also in one or more of `slotRange [t1, t2]` or `slotRange [t2, t3]` if t1 < t2 < t3"

If `t2` corresponds to the start time of slot `s`, and if slot `s` has transactions, then those transactions will indeed occur in both `slotRange t1 t2` and `slotRange t2 t3` (as our range filtering is inclusive).